### PR TITLE
Add the README file in docs folder

### PR DESCRIPTION
**Pull request by XiangyuG**
Add instruction files to explain what is inside each folder of Chipmunk compiler

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Compiler source code organization
+
+```
+
+chipmunk-tofino
+├── chipc
+│   ├── lib                   -- library files including antlr version 4.7.2
+│   └── templates             -- collection of jinja2 template files
+├── chipc.egg-info            -- some egg info
```
Comment (taegyunkim) [inline docs/README.md]: Remove this

https://github.com/chipmunk-project/chipmunk-tofino/blob/2dac70e2b36356102706c23c6c684088f60dae06/.gitignore#L16

*.egg-info directory is ignored by Git, no need to document. 
btw, egg-info contains project metdadata: https://setuptools.readthedocs.io/en/latest/formats.html#eggs-and-their-formats

**Review by taegyunkim — CHANGES_REQUESTED** [review]

**Comment by taegyunkim** [comment]
You can just put this in the top level README.md file. 

FYI, there are tools to auto-generate documentations, https://wiki.python.org/moin/DocumentationTools, but that would require you to write more detailed class/function level comments in certain styles, and It would be too much work for a research project..

File: docs/README.md (hunk 1 above)
Comment (XiangyuG) [reply docs/README.md]: Sure, I will remove it. As for lib folder, is it used to include all dependency library files needed for Chipmunk?

**Comment by XiangyuG** [comment]
> You can just put this in the top level README.md file.
> 
> FYI, there are tools to auto-generate documentations, https://wiki.python.org/moin/DocumentationTools, but that would require you to write more detailed class/function level comments in certain styles, but It would be too much work for a research project..

Thanks for your valuable suggestions. For now, we plan to put them separately and later merge the documentation into the top level README.md file. In addition to description for each folder, usually should we add more documentation sections? For example, https://github.com/p4lang/p4c/blob/master/docs/README.md offers more sections such as **Additional documentation** and **How to contribute** part.

File: docs/README.md (hunk 1 above)
Comment (taegyunkim) [reply docs/README.md]: Yes

**Comment by taegyunkim** [comment]
It's totally up to you. I'm just suggesting that there are ways to build documentations automatically. I'm saying this because if you keep code and its documentation separate, it's very easy for you and others to update documentation whenever you make code changes.

**Comment by XiangyuG** [comment]
> It's totally up to you. I'm just suggesting that there are ways to build documentations automatically. I'm saying this because if you keep code and its documentation separate, it's very easy for you and others to update documentation whenever you make code changes.

Thanks a lot!!! This can definitely help us build better documentations.

**Pull request merged**